tests/test_install.sh

```diff
@@ -1,192 +1,284 @@
   echo "########################################################################################"
   read -r -p 'Uninstall NGINX Ingress Controller [y/n]: '
-
-  if [[ ${REPLY} == 'y' ]]; then
-    helm uninstall ingress-nginx --namespace ${INGRESS_NAMESPACE}
+  if [ $REPLY == 'y' ]; then
+    helm uninstall ingress-nginx-${RELEASE_VER} --namespace ${INGRESS_NAMESPACE}
+    # kubectl delete namespace ${INGRESS_NAMESPACE}
+  else
+    echo "########################################################################################"
+    echo "#  Skipping NGINX Ingress Controller uninstall"
+    echo "########################################################################################"
   fi
 
 elif [[ "$1" == "upgrade" ]]; then
 
+  # Get the Redis password
+  export REDIS_PASSWORD=$(kubectl get secret --namespace ${REDIS_NAMESPACE} redis -o jsonpath="{.data.redis-password}" | base64 -d)
+
   echo "########################################################################################"
   echo "#  Upgrade Codesealer Release"
   echo "########################################################################################"
-
   helm repo update codesealer
-  helm upgrade codesealer ${CODESEALER_HELM_CHART} --namespace codesealer-system \
-    --set codesealerToken="${CODESEALER_TOKEN}" \
-    --set worker.ingress.namespace="${INGRESS_NAMESPACE}" \
+  helm upgrade codesealer-${RELEASE_VER} ${CODESEALER_HELM_CHART} --namespace codesealer-system \
+    --set codesealerToken=${CODESEALER_TOKEN} \
+    --set worker.ingress.namespace=${INGRESS_NAMESPACE} \
+    --set worker.ingress.deployment=${INGRESS_DEPLOYMENT} \
+    --set worker.ingress.port=${INGRESS_PORT} \
+    --set worker.redis.service.name=redis-${RELEASE_VER}-master \
+    --set worker.config.bootloader.redisUser=default \
+    --set worker.config.bootloader.redisPassword=${REDIS_PASSWORD} \
+    --set worker.config.bootloader.redisUseTLS=false \
+    --set worker.config.bootloader.redisIgnoreTLS=true \
     --set worker.config.endpoint.wafMonitorMode=false \
     --set worker.config.endpoint.enableWaf=true \
     --set worker.config.endpoint.wafFullTransaction=true \
-    --set worker.config.endpoint.paranoiaLevel=1
+    --set worker.config.endpoint.crs.paranoiaLevel=1 \
+    --wait --timeout=90s
 
   echo "########################################################################################"
   echo "#  Upgrade Codesealer"
   echo "########################################################################################"
-
   read -r -s -p 'Press any key to continue.'
   kubectl rollout restart deployments --namespace codesealer-system
-  kubectl rollout status deployments --namespace codesealer-system
+
+  echo "########################################################################################"
+  echo "#  Please wait 1 minute..."
+  echo "########################################################################################"
+  sleep 60
```

Comment:
Kubernetes has built in functionality for waiting on resources, why would you do arbitrary 1 minute waits?

Reply:
Agreed.